File: remove-bg.mjs
// Remove white background using Jimp v1 API
import Jimp from 'jimp';

const src = 'public/assets/hope-logo.png';
const WHITE_THRESHOLD = 230;

const image = await Jimp.read(src);
const { width, height, data } = image.bitmap;

for (let y = 0; y < height; y++) {
  for (let x = 0; x < width; x++) {
    const idx = (y * width + x) * 4;
    const r = data[idx];
    const g = data[idx + 1];
    const b = data[idx + 2];
    // Make near-white pixels fully transparent
    if (r > WHITE_THRESHOLD && g > WHITE_THRESHOLD && b > WHITE_THRESHOLD) {
      data[idx + 3] = 0;
    }
  }
}

await image.write(src);
console.log('✅ White background removed and saved to', src);

File: remove_bg.py
from PIL import Image
import numpy as np

img = Image.open('public/assets/hope-logo.png').convert('RGBA')
data = np.array(img)

r, g, b, a = data[:,:,0], data[:,:,1], data[:,:,2], data[:,:,3]

# Remove near-black pixels: R<40, G<40, B<40
black_mask = (r < 55) & (g < 55) & (b < 55)

# Make them transparent
data[:,:,3] = np.where(black_mask, 0, a)

# Also soft-edge near-black (40-90 range) for anti-alias smoothness
near_black = (r < 90) & (g < 90) & (b < 90) & ~black_mask
darkness = np.maximum(np.maximum(r, g), b).astype(float)
# alpha ramp from 0 at 40 to full at 90
alpha_scale = np.clip((darkness - 40) / 50.0, 0, 1)
data[:,:,3] = np.where(near_black, (a * alpha_scale).astype(np.uint8), data[:,:,3])

result = Image.fromarray(data)
result.save('public/assets/hope-logo.png', 'PNG')
print('Done! Logo saved with transparent background.')
